Signing 3rd party library with Microsoft cert in UB build

Public unified build builds are failing to sign an aspnetcore artifact with the following error:


```
##[error]artifacts/source-built-sdks/Microsoft.DotNet.Arcade.Sdk/tools/Sign.proj(76,5): error SIGN004: Signing 3rd party library '/__w/1/s/src/aspnetcore/artifacts/tmp/Release/ContainerSigning/1157/./shared/Microsoft.AspNetCore.App/10.0.0-ci/Microsoft.AspNetCore.App.Runtime.Composite.r2r.dll' with Microsoft certificate 'MicrosoftDotNet500'. The library is considered 3rd party library due to its copyright: ''.
```

[Link to failure](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1042758&view=logs&j=448b980d-0c5b-5119-cc22-42373565c49c&t=68f87380-86e4-5224-4310-49d34dae7c37&l=3147)

--- a/eng/Signing.props
+++ b/eng/Signing.props
@@ -17,6 +17,12 @@
     <FileSignInfo Include="Mono.Cecil.Rocks.dll" CertificateName="3PartySHA2" />
   </ItemGroup>
 
+  <ItemGroup>
+    <!-- Composite r2r images show up as fully native images and trigger the 3rd party warning when signing on Mac/Linux
+         even though they don't have readable copyright info. Skip those files in the signing check. -->
+    <ItemsToSkip3rdPartyCheck Include="Microsoft.AspNetCore.App.Runtime.Composite.r2r.dll" />
+  </ItemGroup>
+
   <PropertyGroup>
     <BaseRedistNetCorePath>$(ArtifactsObjDir)RedistSharedFx.Layout\$(Configuration)\</BaseRedistNetCorePath>
     <RedistNetCorePath>$(BaseRedistNetCorePath)$(TargetRuntimeIdentifier)\</RedistNetCorePath>

